### ISSUE TITLE
Multiday are events not spanning all days

### ISSUE THREAD

**Issue by nathanmay** [omnia-digital/livewire-calendar#31]
Using your demo code for conveneince, when setting a start date and end date, the event only shows on the start date and doesnt span across all days.

I have tried with both the multiday tag and without.

This was previously working but I have recently upgraded to Livewire 3 and laravel 11 - So guessing it is somewhat related.

I am runing php 8.2.

```
public function events() : Collection
{
    return collect([
        [
            'id' => 1,
            'title' => 'Breakfast',
            'description' => 'Pancakes! 🥞',
            'date' => Carbon::today(),
            'end' => Carbon::tomorrow(),
        ],
    ]);
}
```
**OR...**
```
public function events() : Collection
{
    return collect([
        [
            'id' => 1,
            'title' => 'Breakfast',
            'description' => 'Pancakes! 🥞',
            'date' => Carbon::today(),
            'end' => Carbon::tomorrow(),
            'multiday' => true
        ],
    ]);
}
```

<img width="729" alt="Screenshot 2024-05-24 182613" src="https://github.com/omnia-digital/livewire-calendar/assets/33894037/39bcde9a-679c-4ce5-8b5a-19e2af9d949f">

**Comment by lrljoe** [comment]
Yeah, there are definitely bugs with this, there are ways to get this working, but given the number of issues, and staleness, I doubt it'll get resolved.

I've just applied fixes locally on a private fork.

**Comment by joshtorres** [comment]
Thank you for pointing this out and trying to find a resolution. I don't have time to work on this currently but if you can find a solution and create a PR, I will look into merging that. If not, I'll try to look into this in the future. Thanks again.

**Comment by lrljoe** [comment]
@joshtorres  - I do have a solution for this, just need to look at how best to document the behaviour!

**Comment by joshtorres** [comment]
@lrljoe ok awesome. If you want to create a PR when you're ready, I'll look it over.

**Comment by lrljoe** [comment]
@joshtorres  - I'm in two minds about whether it should be a PR, or just some simple documentation to be honest!

At the moment, my (pretty simple) modification allows for:
- Multi Day spanning events
E.g.
Event starts on Monday at 09:00
Runs 24/7 Until:
Event finishes on Wednesday at 17:00

- Multi Day repeated events
E.g.
Event runs for 3 consecutive days, 09:00 - 17:00 each day

To do this smoothly I introduced these properties into the Events collection:
* start_date
* end_date
* start_time
* end_time

The only real change is over-riding getEventsForDay()

I've had to reduce the font-size to fit it clearly into a screenshot, but this demonstrates the behaviour:

![image](https://github.com/user-attachments/assets/56d7334d-40a5-47ba-b7ce-0423b72f9757)

**Comment by lrljoe** [comment]
@joshtorres - what do you think out of the following:

## Option 1
- Docs to explain how to **extend** the default behaviour, to enable:
  - Multi-Day Spanning
  - Repeat Events

OR

## Option 2
- Adding in a secondary Component that extends the base "LivewireCalendar" (e.g. LivewireCalendarExtended) and provides the approaches for:
  - Multi-Day Spanning
  - Repeat Events

Along with the relevant docs

OR

## Option 3
- Adding in backwards-compatible code that enables the new functionality, while not breaking anything existing, which would be a longer term approach, as people will have all kinds of deviations from the norm

**Comment by joshtorres** [comment]
I'm thinking just having Docs might be the best option, depending on how complicated it is. Since everyone won't need that functionality, it's probably not worth our time to reinvent it. If you want to do Option 2 so they have an example, that would be really nice and helpful, but having docs would already make it way better. Thanks!

**Comment by lrljoe** [comment]
Self prompt so it doesn't slip off my radar, I'll look to write the docs this side of xmas
